source/fuzz/transformation_changing_memory_semantics.h

```diff
@@ -0,0 +1,118 @@
+  // instruction
+  //   would like to change, must be equal to 2 or 3 only.
+  // - |message_.memory_semantics_new_value_id| the new id of memory semantics
+  //   that is will change with the old.
+  bool IsApplicable(
+      opt::IRContext* ir_context,
+      const TransformationContext& transformation_context) const override;
+
+  // Change id of memory semantics for specific position with a new one.
+  void Apply(opt::IRContext* ir_context,
+             TransformationContext* transformation_context) const override;
+
+  // Check if instruction is atomic instruction only. Check if Memory semantics
+  // operand Position equal to 0 or 1. Check if operand position equal to 0 in
+  // the case of instruction that takes one memory semantics operand and 0 or 1
+  // for instructions that takes two operands.
+  static bool IsNeededOpcodeWithAppropriatePosition(SpvOp opcode,
+                                                    uint32_t position);
+
+  // - |opcode| opcode must be atomic or barrier instruction.
+  //   Valid memory semantics masks:
+  //   OpAtomicLoad: None | Acquire | SequentiallyConsistent
+  //   OpAtomicStore: None | Release | SequentiallyConsistent
+  //   OpAtomicRMW instructions: None | AcquireRelease | SequentiallyConsistent
+  //   Barrier instructions:  None | Acquire | Release | AcquireRelease |
+  //   SequentiallyConsistent
+  // - |lower_bits_old_memory_semantics| must be suitable for specific atomic
+  //   or barrier instruction.
+  // - |lower_bits_new_memory_semantics| should be larger than the old one,
+  //   must be suitable for specific atomic or barrier instruction.
+  // - |memory_model| return false if the memory model is Vulkan and memory
+  //   semantics is Sequentially Consistent.
+  static bool IsValidConversion(
+      SpvOp opcode, SpvMemorySemanticsMask first_5bits_old_memory_semantics,
+      SpvMemorySemanticsMask first_5bits_new_memory_semantics,
+      SpvMemoryModel memory_model);
+
+  // Check available memory semantics values for the atomic load instruction.
+  static bool IsAtomicLoadMemorySemanticsValue(
+      SpvMemorySemanticsMask memory_semantics_value);
+
+  // Check available memory semantics values for the atomic store instruction.
+  static bool IsAtomicStoreMemorySemanticsValue(
+      SpvMemorySemanticsMask memory_semantics_value);
+
+  // Check available memory semantics values for the atomic read modify write
+  // instructions.
+  static bool IsAtomicRMWInstructionsemorySemanticsValue(
+      SpvMemorySemanticsMask memory_semantics_value);
+
+  // Check available memory semantics values for the barrier instructions.
+  static bool IsBarrierInstructionsMemorySemanticsValue(
+      SpvMemorySemanticsMask memory_semantics_value);
+
+  // Return one if instruction is OpMemoryBarrier, else will return
+  // suitable position value (two or three).
+  static uint32_t GetMemorySemanticsOperandIndex(SpvOp opcode,
+                                                 uint32_t zero_or_one);
```

Comment:
Rename function: GetMemorySemanticsInOperandIndex

Rename parameter: `zero_or_one` -> `memory_semantics_operand_position`

Change comment to: Returns the "in operand" index of the memory semantics operand for the instruction. 
- |opcode|: the instruction opcode.
- |memory_semantics_operand_position|: 0 for the first memory semantics operand or 1 for the second memory semantics operand in the instruction.

Reply:
Done.